scheduler/core/components/optimizer/greedymax.py

```diff
@@ -493,18 +510,23 @@ def nir_slots(self, science_obs, n_slots_filled, len_interval) -> Tuple[int, int
 
         return slot_start_nir, slot_end_nir, obs_id_nir
 
-    def mean_airmass(self, obsid, interval, night=0):
+    def mean_airmass(self, obs_id: ObservationID, interval: Interval, night_idx: NightIndex) -> npt.NDArray[float]:
         """
         Calculate the mean airmass of an observation over the given interval
         """
-        programid = obsid.program_id()
+        programid = obs_id.program_id()
         #     print(obsid.id, programid.id)
-        airmass = self.selection.program_info[programid].target_info[obsid][night].airmass[interval]
+        airmass = self.selection.program_info[programid].target_info[obs_id][night_idx].airmass[interval]
 
         return np.mean(airmass)
 
-    def place_standards(self, night, interval, science_obs, partner_obs, n_std, verbose: bool = True) \
-            -> Tuple[List, List]:
+    def place_standards(self,
```

Comment:
I moved those to Plan (not being used yet) because I imagine with NIR science this should always happen so it should be done at the moment of the insertion

Reply:
👍